spark/src/main/resources/python/zeppelin_pyspark.py

```diff
@@ -365,10 +345,20 @@ def getCompletion(self, text_value):
           mod = ast.Module([node])
           code = compile(mod, '<stdin>', 'exec')
           exec(code, _zcUserQueryNameSpace)
+
+        intp.setStatementsFinished("", False)
       except:
-        raise Exception(traceback.format_exc())
+        exception = traceback.format_exc()
+        m = re.search("File \"<stdin>\", line (\d+).*", exception)
+        if m:
+          line_no = int(m.group(1))
+          intp.setStatementsFinished(
+            "Fail to execute line {}: {}\n".format(line_no, stmts[line_no - 1]) + exception, True)
```

Comment:
line_no is 1 base, you can check the test code where we get the correct line number.
Actually the line number in frontend is not correct, the first line may be the interpreter indicator.
See
![image](https://cloud.githubusercontent.com/assets/164491/25382015/7b73c28a-29e8-11e7-83ca-8618d5f26260.png)